module/VuFind/src/VuFind/Log/LoggerFactory.php

```
@@ -367,12 +368,11 @@ protected function addHandlers(
             // Ensure verbosity is an int, default to 1 if not specified or invalid
             $verbosity = isset($parts[1]) && is_numeric($parts[1]) ? (int)$parts[1] : 1;
 
-            $min = LogLevel::DEBUG; // Default min, will be overwritten by switch
-            $max = LogLevel::EMERGENCY; // Default max, will be overwritten by switch
```

Why did you remove these? It appears that without them, there is a possibility that a FilterHandler will be passed uninitialized values in line 421 when an unexpected level value is passed in the configuration (unless I'm misreading something).

Maybe we need a test case that passes a gibberish level to see what happens (and capture our expectations in a test).

---

The switch/case overrides them or does a `continue 2` in all cases. I can put them back, but they don't serve a purpose. PHPStan would catch any errors in the switch/case (I tested this!). Or to put it the other way: if they're always supposed to be overridden, having no defaults allows PHPStan to detect the problem that would otherwise go undetected.

---

I suppose another option is to add the defaults back and let the switch-case fall through and use them if the priority is invalid. That might be a better default than silently dropping the option.

---

Ahh, sorry, I was misunderstanding where the `continue 2` would direct the code flow.

Maybe it does make sense to use the defaults if an unexpected log level is provided, since that will then allow us to send a log message to the configured provider complaining about an unexpected log level. :-)

---

Done, apart from logging the bad logging level. But now it'd be possible!

---

Thanks. I think before we merge this we should either add logging of the bad logging level, or else we should add a note to the configuration settings saying that unrecognized level values will cause full logging. (I'd favor actually adding the log message -- might as well do this the whole way while we're here).

---

Okay, I'll add it.

---

Done!